src/main/webapp/js/datahandlers/scholarshipdatahandler.js

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** @fileoverview This class loads scholarship data from the servlet. */
+
+goog.module('datahandlers.scholarshipdatahandler');
+const {Map: SoyMap} = goog.require('soy.map');
+
+/** This class loads scholarship data from the backend and formats it for soy templates.  */
+class ScholarshipDataHandler {
+
+  /** 
+   * @return The string path of the endpoint to send a fetch request to. 
+   * @private 
+   */
+  get scholarshipEndPoint_() {
+    return '/scholarship-data';
+  }
+
+  /**
+   * This method converts from scholarship JSON object to a JS object map, 
+   *  which will be used to render the scholarship page soy template.
+   * @param {*} data - The JSON object to be converted.
+   * @return {Object} - The object map representing a scholarship's data.
+   * @private
+   */
+  async convertFromJsonToTemplate_(data) {
+    const SEPARATOR = ', ';
+    const requirementsMap = new Map();
```

Comment:
Since all requirements has the same format, we can make a map in js, set requirement titles to be the keys and use a for loop to render all requirements at once.